main.ino

```diff
@@ -73,6 +75,30 @@ void loop()
       
         Serial.println();
     }
+    
+    unsigned long timeSinceStart = millis();
+    if (!lastStatusCheckTime
+        || timeSinceStart < lastStatusCheckTime // millis() overflow
```

Comment:
Well done millis rollover!
You are in 0.001% of people that read and actually understand documentation :-)